node-test/src/test/scala/org/bitcoins/node/NeutrinoNodeWithWalletTest.scala

```diff
@@ -36,26 +34,109 @@ class NeutrinoNodeWithWalletTest extends NodeUnitTest {
                                          Some(BitcoindVersion.Experimental))
   }
 
-  private val assertionP: Promise[Boolean] = Promise()
-
-  private val walletP: Promise[UnlockedWalletApi] = Promise()
-  private val walletF: Future[UnlockedWalletApi] = walletP.future
+  private var walletP: Promise[UnlockedWalletApi] = Promise()
+  private var walletF: Future[UnlockedWalletApi] = walletP.future
+  after {
+    //reset assertion after a test runs, because we
+    //are doing mutation to work around our callback
+    //limitations, we can't currently modify callbacks
+    //after a NeutrinoNode is constructed :-(
+    walletP = Promise()
+    walletF = walletP.future
+  }
 
-  val amountFromBitcoind = 1.bitcoin
+  val OneBitcoin = 1.bitcoin
+  val FeeRate = SatoshisPerByte(10.sats)
+  val Fees = 2240.sats
 
-  def callbacks: SpvNodeCallbacks = {
+  def callbacks: NodeCallbacks = {
     val onBlock: DataMessageHandler.OnBlockReceived = { block =>
       for {
         wallet <- walletF
-        _ <- wallet.processBlock(block, confirmations = 0)
+        _ <- wallet.processBlock(block, confirmations = 6)
+      } yield ()
+    }
+    val onCompactFilter: OnCompactFilterReceived = { (blockHash, blockFilter) =>
+      for {
+        wallet <- walletF
+        _ <- wallet.processCompactFilter(blockHash, blockFilter)
       } yield ()
     }
-    SpvNodeCallbacks(
-      onBlockReceived = Seq(onBlock)
+
+    NodeCallbacks(
+      onBlockReceived = Seq(onBlock),
+      onCompactFilterReceived = Seq(onCompactFilter)
     )
   }
 
   it must "receive information about received payments" taggedAs (UsesExperimentalBitcoind) in {
+    param =>
+      val NeutrinoNodeFundedWalletBitcoind(node, wallet, bitcoind) = param
+
+      walletP.success(wallet)
+
+      def condition(
+          expectedConfirmedAmount: CurrencyUnit,
+          expectedUnconfirmedAmount: CurrencyUnit,
+          expectedUtxos: Int,
+          expectedAddresses: Int): Future[Boolean] = {
+        for {
+          confirmedBalance <- wallet.getConfirmedBalance()
+          unconfirmedBalance <- wallet.getUnconfirmedBalance()
+          addresses <- wallet.listAddresses()
+          utxos <- wallet.listUtxos()
+        } yield {
+          (expectedConfirmedAmount == confirmedBalance) &&
+          (expectedUnconfirmedAmount == unconfirmedBalance) &&
+          (expectedAddresses == addresses.size) &&
+          (expectedUtxos == utxos.size)
+        }
+      }
+
+      val condition1 = () => {
+        condition(0.sats,
```

Comment:
nit: can you use named arguments (`alt+enter` on your keyboard in intellij). This makes this kind of stuff so much more readable for code reviewers :-)